### PR TITLE
Default end date for pynance.data.get

### PR DESCRIPTION
**Pull request by arkadasgupta**

### CODE REVIEW

File: pynance/data/retrieve.py [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
-
+from datetime import datetime
 from ftplib import FTP
 from functools import partial
 import io
 
 import pandas_datareader.data as web
 
-def get(equity, start, end):
+def get(equity, start, end = datetime.today().date().isoformat()):
```
Comment (aisthesis) [inline pynance/data/retrieve.py]: This should be end=None. Then test inside the function for None and replace it. 

The default value should be at least 1 day after today in order to account for potential time zone differences. It doesn't even matter if it's a long time afterward (e.g., '2017' right now). 

For performance purposes, I'd look at the type of time object that is actually being used under the hood (pd.Timestamp, I think) and use that data type. Then there's no unnecessary conversion from one type to another.

Comment (arkadasgupta) [inline pynance/data/retrieve.py]: I am not sure what you mean by the last comment. Are you saying we change the iso dates into pd.Timestamp before calling the pd api?

Comment (aisthesis) [inline pynance/data/retrieve.py]: The iso formatted date just gets changed back into a datetime object anyway:

- https://github.com/pydata/pandas-datareader/blob/823e6fc6b82c6d1cd5645bdc4da87094e7773f51/pandas_datareader/yahoo/daily.py#L60
- https://github.com/pydata/pandas-datareader/blob/823e6fc6b82c6d1cd5645bdc4da87094e7773f51/pandas_datareader/yahoo/daily.py#L60

I also noticed that the DataReader object allows end (and even start) to be None:

    >>> df = web.DataReader('amzn', 'yahoo', '2016-02-01')
    >>>

The best way might be just to let the DataReader handle it:

   if end == None:
       return web.DataReader(equity, 'yahoo', start)
   ...

Comment (aisthesis) [inline pynance/data/retrieve.py]: Another option worth considering: Change the signature to `get(equity, *args)` and have the body be:

    return web.DataReader(equity, 'yahoo', *args)

I think that would work, and it's clearly simplest. Then we could also show in the documentation that start defaults to None.

Comment (aisthesis) [inline pynance/data/retrieve.py]: For documentation methods see:

- http://stackoverflow.com/questions/12082570/override-function-declaration-in-autodoc-for-sphinx/12087750#12087750
- http://stackoverflow.com/questions/5365684/is-it-possible-to-override-sphinx-autodoc-for-specific-functions

Comment (arkadasgupta) [inline pynance/data/retrieve.py]: the args kwargs approach would require changing calling classes wouldn't they? that is investor-bot code.

Comment (aisthesis) [inline pynance/data/retrieve.py]: shouldn't. What calls are you thinking might not work?

Comment (aisthesis) [inline pynance/data/retrieve.py]: One other thing I wanted to mention: Be sure to put a note in the documentation that this is `new in 0.5.0`, which will be the new version.

Comment (arkadasgupta) [inline pynance/data/retrieve.py]: my bad. in this case it won't require changing the calling classes. i had a different understanding of args-kwargs.

File: pynance/data/retrieve.py [hunk 2]
```diff
@@ -10,24 +10,27 @@
...
 
 import pandas_datareader.data as web
 
-def get(equity, start, end):
+def get(equity, *args, **kwargs):
     """ 
     Get DataFrame for an individual equity from Yahoo!  
-    
+
+    .. changed:: 0.5.0 - changed to variadic parameters
```
Comment (aisthesis) [inline pynance/data/retrieve.py]: The variadic parameters shouldn't be exposed to the casual user. More useful would be something like 'fill in missing start and end dates with defaults'.

File: pynance/data/retrieve.py [hunk 3]
```diff
@@ -10,24 +10,27 @@
...
+from datetime import datetime
 from ftplib import FTP
 from functools import partial
 import io
 
 import pandas_datareader.data as web
 
-def get(equity, start, end):
+def get(equity, *args, **kwargs):
     """
```
Comment (aisthesis) [inline pynance/data/retrieve.py]: The first documentation line needs to be:

    """get(equity, start=None, end=None)

http://stackoverflow.com/questions/5365684/is-it-possible-to-override-sphinx-autodoc-for-specific-functions

File: pynance/data/retrieve.py [hunk 4]
```diff
@@ -10,24 +10,29 @@
...
 import io
 
 import pandas_datareader.data as web
 
-def get(equity, start, end):
+def get(equity, *args, **kwargs):
     """ 
-    Get DataFrame for an individual equity from Yahoo!  
-    
+    get(equity, start = None, end = None)
```
Comment (aisthesis) [inline pynance/data/retrieve.py]: The convention is not to have space around '=' in the function definition. 'start=None', not 'start = None'

File: pynance/data/retrieve.py [hunk 5]
```diff
@@ -10,24 +10,29 @@
 Wraps `Pandas Remote Data Access 
 <http://pandas.pydata.org/pandas-docs/stable/remote_data.html>`_.
 """
-
+from datetime import datetime
```
Comment (aisthesis) [inline pynance/data/retrieve.py]: This line should be removed, and I think the space is important for generating the documentation.

**Comment by aisthesis** [comment]
The test work fine, but the Sphinx documentation build is failing.

**Comment by aisthesis** [comment]
Just fyi, as the Sphinx build errors can be tricky to figure out: http://stackoverflow.com/questions/11874716/sphinx-import-error-in-not-to-document-package-is-there-any-way-to-skip

**Comment by aisthesis** [comment]
Closing pull request to allow debugging on new `develop` branch.

**Pull request closed without merging**